src/hip_hcc.cpp

```diff
@@ -304,18 +304,28 @@ void ihipStream_t::wait(LockedAccessor_StreamCrit_t& crit) {
 
 //---
 // Wait for all kernel and data copy commands in this stream to complete.
-void ihipStream_t::locked_wait() {
+inline void ihipStream_t::locked_wait(bool& waited) {
```

Comment:
Would defining a new function such as `locked_waited()` to return `bool` break compatibility, for example,
`bool ihipStream_t::locked_waited() {  // returns true when waited on a non-empty stream`
could be used instead of overloading `locked_wait()` and avoid defining temporary `bool`s to get result inside `for` loops etc. 
Was there a reason to overload `locked_wait()` instead of defining a new function?

Reply:
it's to avoid breaking compatibility and you can't overload by the return type

Reply:
@scchan Not sure extending the interface by adding a new function and using it in the new/modified logic would break compatibility - compatibility with what?

Reply:
I'm not sure if I understand your question correctly.  What I meant was that having both `void locked_waited()` and `bool locked_waited()` doesn't work

Reply:
@scchan The suggestion was to create a new function `bool locked_wait**ed**()` (not `wait` but `waited` as in waited?), not overloading. (Sorry the names are similar, so easy to have missed that.)

Reply:
sorry I missed the name difference but I still prefer having the overloaded version to show that both are semantically identical and that one gives you the status.

Reply:
@scchan Ok, I understand your preference.
I think the use of overload here, and the resulting code constructs of requiring the use of a temporary variable instead of checking the return value from a call makes the code complicated/less readable. Good to know that there is no compatibility issue preventing the addition of a new function should that approach be considered later.